ilichev_iurii/graph.cpp

```diff
@@ -1,78 +1,85 @@
 #include "graph.hpp"
 
-struct Graph::Edge {
- public:
-  Edge(EdgeId id, VertexId from_vertex_id, VertexId to_vertex_id)
-      : id_(id), from_vertex_id_(from_vertex_id), to_vertex_id_(to_vertex_id) {}
+#include <cassert>
+#include <iostream>
 
-  EdgeId id() const { return id_; }
-  VertexId from_vertex_id() const { return from_vertex_id_; }
-  VertexId to_vertex_id() const { return to_vertex_id_; }
+void Graph::add_vertex() {
+  const auto next_vertex_id = VertexId(vertexes_.size());
 
-  bool operator<(const Edge& edge) const { return edge.id_ > id_; }
+  vertexes_.emplace_back(next_vertex_id);
+  vertex_edges_.insert({next_vertex_id, {}});
+}
 
- private:
-  EdgeId id_ = 0;
-  VertexId from_vertex_id_ = 0;
-  VertexId to_vertex_id_ = 0;
-};
+void Graph::add_edge(VertexId from_vertex_id, VertexId to_vertex_id) {
+  // Checking that an edge can be added between these vertices
+  assert(edge_correctness(from_vertex_id, to_vertex_id));
```

Comment:
Вот оно, ты все проверки смешал в одну кучу, это тебе не поможет дебажить код. Смысл в том, чтобы каждая проверка тестировала минимальную логику, чтобы можно было быстро и легко выявить ошибку.
```suggestion
  assert(has_vertex(from_vertex_id));
  assert(has_vertex(to_vertex_id));
  assert(!has_edge(from_vertex_id, to_vertex_id));
```